rules-java-archives/addon/src/main/java/org/jboss/windup/rules/apps/java/archives/identify/SortedFileArchiveIdentifier.java

```diff
@@ -0,0 +1,264 @@
+package org.jboss.windup.rules.apps.java.archives.identify;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.RandomAccessFile;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.regex.Pattern;
+import org.apache.commons.lang.StringUtils;
+import org.jboss.forge.addon.dependencies.Coordinate;
+import org.jboss.forge.addon.dependencies.builder.CoordinateBuilder;
+import org.jboss.windup.rules.apps.java.archives.identify.api.ArchiveHashIdentifier;
+
+/**
+ * Searches a sorted file with lines of this format:
+ * "SHA1_hash G:A:V".
+ *
+ * Initially, no mapping is done. As this identifier looks for the values
+ * and jumps around the file, the hashes and their position is added to a
+ * hash -> positions cache map (posCache). This is then used in further searches.
+ *
+ * @author Ondrej Zizka, ozizka at redhat.com
+ */
+public class SortedFileArchiveIdentifier implements ArchiveHashIdentifier
+{
+    public static final int SHA1_LENGTH = 40;
+
+
+    // Input
+    private File dataFile;
+
+    // Work variables
+    private final long dataSize;
+    private FileInputStream dataFileIS = null;
+
+    // SHA1 hash -> file position where it is.
+    private SortedMap<String, Long> posCache = new TreeMap();
+
+
+    public SortedFileArchiveIdentifier(File dataFile) throws FileNotFoundException
+    {
+        if (!dataFile.exists())
+            throw new IllegalArgumentException("Hash to GAV data file does not exist: " + dataFile.toString());
```

Comment:
Wouldn't this prevent the constructor from actually throwing FileNotFoundException?